WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -220,6 +220,7 @@ final class ProductSelectorViewModel: ObservableObject {
                                                  selectedProductVariationIDs: selectedItems,
                                                  purchasableItemsOnly: purchasableItemsOnly,
                                                  supportsMultipleSelection: supportsMultipleSelection,
+                                                 isClearSelectionEnabled: toggleAllVariationsOnSelection,
```

Comment:
Is using `toggleAllVariationsOnSelection` here a side-effect? Possibly fine but I don't 100% see why we don't pass the setting to this view model too

Reply:
> Is using toggleAllVariationsOnSelection here a side-effect? Possibly fine but I don't 100% see why we don't pass the setting to this view model too

Good question! At the moment `ProductVariationSelectorViewModel` is instantiated within `ProductSelectorViewModel`, and this view model is used in two different places: Order Creation, and Coupons.

If we set this setting directly here we won't be able to use different behaviour for Products and Coupons when dealing with Product Variations, so we pass it when we create the different view models for product lists for each feature instead, in this case on [EditableOrderViewModel](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/EditableOrderViewModel.swift#L177)

Reply:
@iamgabrielma Sorry, I'm still a bit unsure...

> If we set this setting directly here we won't be able to use different behaviour for Products and Coupons when dealing with Product Variations, so we pass it when we create the different view models for product lists for each feature instead, in this case on EditableOrderViewModel

Yes, understood and I agree that we need to pass in the `isClearSelectionEnabled` to the `ProductVariationSelectorViewModel`, so that it can behave correctly in both contexts 👍 

**However** my question was about what we're using to set that flag:

Here we're using `toggleAllVariationsOnSelection` to set the `isClearSelectionEnabled` value on the view model we're creating. Toggling all variations, and clearing selection, seem like two different things. Perhaps they're only subtly different... but if they're different at all, we should perhaps use some explicit logic rather than relying on a side effect?

(The side effect that I'm talking about is "If we don't toggle all variations on selection, we shouldn't allow the selection to be cleared".)

I may still be misunderstanding... sorry if so!

Reply:
> Toggling all variations, and clearing selection, seem like two different things. Perhaps they're only subtly different... but if they're different at all, we should perhaps use some explicit logic rather than relying on a side effect?
> (The side effect that I'm talking about is "If we don't toggle all variations on selection, we shouldn't allow the selection to be cleared".)

Oh I see what you mean now, I miss-understood the initial comment. Right now I'm indeed relying on the side-effect of " If we cannot toggle variations, we shouldn't have the clear selection button enabled either", but this won't be necessarily true moving forward, neither seems "right" and should be decoupled and made more explicit. I'll address this on https://github.com/woocommerce/woocommerce-ios/issues/9043